private/bufpkg/bufcheck/bufbreaking/internal/bufbreakingcheck/bufbreakingcheck.go

```diff
@@ -291,11 +291,17 @@ func checkFieldSameOneof(add addFunc, corpus *corpus, previousField protosource.
 	if !previousInsideOneof && !insideOneof {
 		return nil
 	}
+	previousProto3Optional := previousOneof != nil && previousOneof.IsProto3OptionalSyntheticOneof()
+	proto3Optional := oneof != nil && oneof.IsProto3OptionalSyntheticOneof()
 	if previousInsideOneof && insideOneof {
 		if previousOneof.Name() != oneof.Name() {
-			// otherwise prints as hex
-			numberString := strconv.FormatInt(int64(field.Number()), 10)
-			add(field, nil, field.Location(), `Field %q on message %q moved from oneof %q to oneof %q.`, numberString, field.Message().Name(), previousOneof.Name(), oneof.Name())
+			// If we are inside a oneof, then this means that we renamed the field, which is
+			// a different check.
+			if !previousProto3Optional && !proto3Optional {
+				// otherwise prints as hex
```

Comment:
What is this comment? I know it came from the original source, but I don't know what it means.

Reply:
~~`field.Number()` is an `int`, why would `%d` turn it into a hex?~~
Figured it out, it's the `%q` that's interpreting it as a character-literal. Could just do `"%d"`, but I digress.